.github/actions/test/integration/test/cloudhypervisor/action.yml

```diff
@@ -0,0 +1,133 @@
+        USER="root"
+        IP_HV1="192.168.122.2"
+        IP_HV2="192.168.122.3"
+
+        scp $SSH_OPTS ./.github/actions/test/integration/test/cloudhypervisor/vm.xml "$USER@$IP_HV1:/opt/vm-chv.xml"
+        scp $SSH_OPTS /opt/ubuntu.raw "$USER@$IP_HV1:/var/lib/libvirt/images/ubuntu.raw"
+        scp $SSH_OPTS /opt/ubuntu-cloud-init-ds.iso "$USER@$IP_HV1:/var/lib/libvirt/images/ubuntu-cloud-init-ds.iso"
+
+        # the cloud-init data is not migrated from source to destination, but needed for the VM to run at the destination
+        scp $SSH_OPTS /opt/ubuntu-cloud-init-ds.iso "$USER@$IP_HV2:/var/lib/libvirt/images/ubuntu-cloud-init-ds.iso"
+
+    - name: Start VM on HyperVisor 1
+      shell: bash
+      run: |
+        KEY=/opt/ssh_host_ed25519_key
+        SSH_OPTS="-i $KEY -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null"
+        USER="root"
+        IP_HV1="192.168.122.2"
+
+        ssh $SSH_OPTS "$USER@$IP_HV1" << EOF
+          virsh -c ch:///session define /opt/vm-chv.xml
+          virsh -c ch:///session start VM
+        EOF
+
+    - name: Wait for VM to be ready
+      shell: bash
+      run: |
+        KEY=/opt/ssh_host_ed25519_key
+        SSH_OPTS="-i $KEY -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null"
+        USER="root"
+        IP_HV1="192.168.122.2"
+
+        ssh $SSH_OPTS "$USER@$IP_HV1" << EOF
+          KEY=/opt/ssh_host_ed25519_key
+          SSH_OPTS="-i \$KEY -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null"
+          USER="root"
+
+          MAX_ITER=40
+          for ((i=1; i<=MAX_ITER; i++)); do
+            IP=\$(virsh -c ch:///session domifaddr VM | awk '/ipv4/ {print \$4}' | cut -d'/' -f1)
+            if [ -z "\$IP" ]; then
+              echo "No IP address found for VM. Retrying..."
+            elif ssh \$SSH_OPTS "\$USER@\$IP" 'exit' 2>/dev/null; then
+              echo "VM is up at IP: \$IP"
+              # create a dummy file for verification later
+              ssh \$SSH_OPTS "\$USER@\$IP" 'echo "Hello, World!" > /opt/hello.txt'
+              break
+            fi
+            if (( i == MAX_ITER )); then
+              echo "Timeout waiting for VM to respond to SSH."
+              cat /var/log/CHV.log
+              exit 1
+            fi
+            echo "Waiting for VM to respond to SSH... (\$i/\$MAX_ITER)"
+            sleep 10
+          done
+        EOF
+
+      # TODO not sure if the connection string actually is ch+ssh://USER@IP/system
+    - name: Mirgrate the VM to HyperVisor 2
+      shell: bash
+      run: |
+        KEY=/opt/ssh_host_ed25519_key
+        SSH_OPTS="-i $KEY -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null"
+        USER="root"
+        IP_HV1="192.168.122.2"
+
+        ssh $SSH_OPTS "$USER@$IP_HV1" << EOF
+          USER="root"
+          IP_HV2="192.168.122.3"
+
+          virsh -c ch:///session migrate --persistent --undefinesource --copy-storage-all --live VM ch+ssh://\$USER@\$IP_HV2/system
```

Comment:
Seems like the live migration using a patched libvirt is not working yet.
We could skip the migration under CHV until it is ready... ?

Reply:
Only works without storage migration for the moment. Not relevant for our HV loads since they use network storage anyway.

Reply:
Having sais that, we should consider hosting these test workloads with network storage as well.